Clarified req_wl value (#360)
oauth-wg/oauth-transaction-tokens: draft-ietf-oauth-transaction-tokens.md

: RECOMMENDED A JSON object that describes the environmental context of the requested transaction. For additional context please refer to {{transaction-context}}

`req_wl`:

> : REQUIRED. A string value that identifies the workload that requested the Txn-Token. If a subsequent transaction token is created from a call to the TTS where the subject token is itself a transaction token, the TTS should maintain the call chain of requesting workloads in the subsequently issued transaction token. The mechanisms for managing this delegation chain is out of scope for this specification.

Reviewer: ```suggestion
: REQUIRED. A string value that identifies the workload requesting the Txn-Token. When the subject token contains information about the Call Chain, including identifiers for other participants in that Call Chain, the information is preserved in the issued Txn-Token. The mechanism for representing and managing the delegation chain is outside the scope of this specification.
```

> : REQUIRED. An string value that identifies the workload that requested the Txn-Token. If a subsequent transaction token is created from a call to the TTS where the subject token is itself a transaction token, the TTS should maintain the call chain of requesting workloads in the subsequently issued transaction token. The mechanisms for managing this delegation chain is out of scope for this specification.

Reviewer: ```suggestion
: REQUIRED. A string value that identifies the workload that requested the Txn-Token. If a subsequent transaction token is created from a call to the TTS where the subject token is itself a transaction token, the TTS should maintain the call chain of requesting workloads in the subsequently issued transaction token. The mechanisms for managing this delegation chain is out of scope for this specification.
```

Reviewer: I wonder if this can be separated from whether a transaction token is used as subject_token. There are other tokens that may have chained identity information as well (as described in Karl's drafts). How about the following:

```suggestion
: REQUIRED. A string value that identifies the workload requesting the Txn-Token. When the subject token contains information about the call chain, including identifiers for other participants in that chain, the TTS should include this information in the issued Txn-Token. The mechanism for representing and managing the delegation chain is outside the scope of this specification.
```

Reviewer: Given wher this is landing and the source of the concerns, why not just simplify to this:

```suggestion
: REQUIRED. A string value that identifies the workload that requested the Txn-Token. 
```

We can keep the text in the "transaction token as subject token" section, but we break the link here. Feels cleaner and also addresses Brian's other concerns about encouraging the use of replacement transaction tokens.

Reviewer 2: I'm good with that. I think the explanation should go in the transaction token as subject token section.

> : REQUIRED. A string or array value that identifies the workload that requested the Txn-Token. In the common special case when there is one requesting workload, the `req_wl` value MAY be a single case-sensitive string. In the general case, when a Txn-Token is replaced by another Txn-Token, the `req_wl` value is an array of case-sensitive strings, identifying subsequent workloads that have requested the Txn-Token. See Section {{replacement}} for more information.

Reviewer: ```suggestion
: REQUIRED. A string or array value that identifies the workload that requested the Txn-Token. In the common special case when there is one requesting workload, the `req_wl` value MAY be a single case-sensitive string. In the general case, when a new Txn-Token is requested using an existing Txn-Token as a `subject_token`, the `req_wl` value is an array of case-sensitive strings, identifying subsequent workloads that have requested the Txn-Token. See Section {{replacement}} for more information.
```

Reviewer 2: Do we need to change this text at all?

> : REQUIRED. A string or array value that identifies the workload that requested the Txn-Token. In the common special case when there is one requesting workload, the `req_wl` value MAY be a single case-sensitive string. In the general case, when a new Txn-Token is requested using an existing Txn-Token as a `subject_token`, the `req_wl` value is an array of case-sensitive strings, identifying subsequent workloads that have requested the Txn-Token. See Section {{replacement}} for more information.

Reviewer 2: ```suggestion
: REQUIRED. An array value that identifies the workload(s) that requested the Txn-Token. In the common case when there is one requesting workload, the `req_wl` value MAY be a single case-sensitive string. In the general case, when a new Txn-Token is requested using an existing Txn-Token as a `subject_token`, the `req_wl` value is an array of case-sensitive strings, identifying subsequent workloads that have requested the Txn-Token. See Section {{replacement}} for more information.
```

Reviewer: Thanks George.

* MAY enable additional asserted values.
* MUST NOT enable modification to asserted values that expand the scope of permitted actions.
* MUST NOT modify `txn`, `sub`, and `aud` values of the Txn-Token in the request.
* MUST NOT issue a new Txn-Token when the Txn-Token being replaced has expired.
* MAY issue a replacement Txn-Token with a lifetime exceeding the lifetime of the input Txn-Token, subject to the policy of the TTS.
* SHOULD limit the number of times a Txn-Token is replaced if it allows extending the lifetime beyond that of the input Txn-Token to reduce replay risks.

> * MUST maintain the Call Chain of workloads that requested the Txn-Token being replaced, and MUST NOT remove or obscure any of those workload identities in the subsequently issued Txn-Token. The mechanism for maintaining this Call Chain is out of scope for this specification.

Reviewer: I would propose dropping "obscure" and just say that is should be maintained (I can imagine some anonymisation may be applied etc, so lets not preclude that and also not introduce a debate about what "obscure" means.

```suggestion
* MUST maintain the Call Chain of workloads that requested the Txn-Token being replaced in the subsequently issued Txn-Token. The mechanism for maintaining this Call Chain is out of scope for this specification.
```

* MUST NOT issue a new Txn-Token when the Txn-Token being replaced has expired.
* MAY issue a replacement Txn-Token with a lifetime exceeding the lifetime of the input Txn-Token, subject to the policy of the TTS.
* SHOULD limit the number of times a Txn-Token is replaced if it allows extending the lifetime beyond that of the input Txn-Token to reduce replay risks.

> * When the subsequent transaction token is created, the TTS should maintain the call chain of requesting workloads in the subsequently issued transaction token. The mechanisms for managing this delegation chain is out of scope for this specification.

Reviewer: ```suggestion
* The TTS should include the Call Chain of requesting workloads in the subsequently issued transaction token. The representation of this Call Chain chain is out of scope for this specification.
```
